sisyphus/task.py

```diff
@@ -209,6 +215,64 @@ def run(self, task_id, resume_job=False, logging_thread=None):
             sys.stderr.flush()
             logging.info("Job finished successful")
 
+    def _wait_for_input_to_sync(self):
+        """
+        Waits for the input files of this task to be synced across the network, eventually raising a
+        TimeoutError.
+
+        The input files are either the ouput files of other jobs or the output files of a preceeding
+        task of this job
+
+        """
+        # Collect expected file sizes, either from a preceeding task or from other jobs
+        logging.info("Getting expected input sizes ...")
+
+        expected_sizes = {}
+        prev = self.get_prev_task()
+
+        if prev:
+            expected_sizes = job.Job._sis_get_expected_file_sizes(self._job, task=prev.name())
+            if expected_sizes is None:
+                logging.warning("This tasks job has already been cleanup up, shouldn't happen!")
+                expected_sizes = {}
+        else: 
+            for i in self._job._sis_inputs:
+                if not i.creator:
+                    logging.info("Cannot check the size of %s, it's not created by sisyphus.", i)
+                    continue
+
+                other_job_sizes = job.Job._sis_get_expected_file_sizes(i.creator)
+                # If the job has been cleaned up, no size info is available, but we can safely
+                # assume that enough time has passed so that all files are synced.
+                if other_job_sizes:
+                    expected_sizes[i.rel_path()] = other_job_sizes[i.rel_path()]
```

Comment:
This fails if a path is only used as prefix without representing a real file. We could change it to something like this:
```
try:
    expected_sizes[rel_path] = other_job_sizes[rel_path]
except KeyError:
    for k, v in other_job_sizes.items():
        if k.startswith(rel_path):
            expected_sizes[k] = v
```
This could also be used if the path is pointing to a directory.